backend_plugin/src/main/evaluators/FeatureEvaluator.java

```diff
@@ -18,7 +18,17 @@
 	public boolean evaluateDocumentChanges(DocumentEvent docEvent) {
 		return false;
 	}
-	
+
+	/**
+	 * Evaluates what changes will be made to the text within a document before the changes are applied
+	 * @param docEvent The DocumentEvent containing the changes about to be made
+	 * @return true if the document changes that are about to occur cause the evaluation function to trigger,
+	 * 		false otherwise.
+	 */
+	public boolean evaluateDocumentAboutToBeChanged(DocumentEvent docEvent) {
```

Comment:
We might want to consider renaming these down the road to something like "evaluateDocumentBeforeChange" and "evaluateDocumentAfterChange" for consistency, but we can worry about that in the coming weeks

Reply:
Great idea. I'll rename the one I created and we can rename the other one when we do some refactoring